### PR TITLE
Add Signal Interference mapper

### PR DESCRIPTION
**Pull request by heyagosper**
Mapper and tests
Also added new mapper to SiteLogmapper and tests.
requires change to ogc-schemas in PR10

### CODE REVIEW

File: src/main/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/SiteLogMapper.java [hunk 1]
```diff
@@ -120,20 +129,20 @@ public A convertFrom(B b, Type<A> type, MappingContext mappingContext) {
...
      * set of domain model equipment log items.
      */
     private <P extends GMLPropertyType, T extends AbstractGMLType, L extends LogItem>
-    BidirectionalConverter<List<P>, Set<L>> equipmentCollectionConverter(Iso<T, L> equipmentIso) {
+    BidirectionalConverter<List<P>, Set<L>> GMLPropertyTypeConverter(Iso<T, L> equipmentIso) {
 
         Iso<P, T> propertyIso = new GMLPropertyTypeMapper<>();
         Iso<P, L> elementIso = propertyIso.compose(equipmentIso);
         return new IsoConverter<>(new ListToSet<>(elementIso));
     }
```
Comment (lbodor) [inline src/main/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/SiteLogMapper.java]: The javadoc and local variables still refer to "equipment". They should now refer to "log items". The method name should start with a lowercase letter. I suggest logItemsConverter.

File: src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/SignalObstructionMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                .getValue();
+
+        SiteLogType siteLog = GeodesyMLUtils.getElementFromJAXBElements(mobs.getElements(), SiteLogType.class)
+                .findFirst().get();
+
+        BasePossibleProblemSourcesType signalObstructionTypeA =
+                siteLog.getSignalObstructionsSet().get(0).getSignalObstructions();
+
+        SignalObstructionLogItem logItem = mapper.to(signalObstructionTypeA);
+        assertEquals(logItem.getPossibleProblemSource(), signalObstructionTypeA.getPossibleProblemSources());
```
Comment (oehm-smith) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/SignalObstructionMapperTest.java]: Asserts back the front.  For TestNG it is assertEquals(actual, expected).  Alternatively use Hamcrest so you can write:

    AssertThat(actual, is(equalTo(expected)) 

to make things more explicit and also avoids the "am I using TestNG or JUnit" conumdrum (both support Hamcrest).

Comment (lbodor) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/SignalObstructionMapperTest.java]: I also think that using Hemcrest to hide junit vs testng conundrum looks like a good idea.

Comment (heyagosper) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/SignalObstructionMapperTest.java]: I also think what everyone else thinks. please create a Jira to change all the everything

File: src/main/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/SiteLogMapper.java (hunk 1 above)
Comment (heyagosper) [inline src/main/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/SiteLogMapper.java]: thanks, I have incorporated these suggestions in the PR

**Pull request merged**